src/Storefront/Page/Navigation/Error/ErrorPageLoader.php

```diff
@@ -50,19 +50,19 @@ public function __construct(
      * @throws MissingRequestParameterException
      * @throws PageNotFoundException
      */
-    public function load(string $cmsErrorLayoutId, Request $request, SalesChannelContext $context): ErrorPage
+    public function load(?string $cms404ErrorLayoutId, Request $request, SalesChannelContext $context): ErrorPage
```

Comment:
Signatures of public methods shouldn't be changed directly like this, we'd need to deprecate the current form beforehand. Would you mind going into detail on the reason for this change a bit more? As far as I can tell, this would lead to a more or less generic method being very specific to the one use-case described in this PR. I think a less intrusive solution could be possible here.